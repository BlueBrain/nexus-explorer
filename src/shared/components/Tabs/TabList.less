.tab-item {
	display: flex;
	flex-direction: column;
	justify-content: flex-start;
	text-align: left;
	max-width: 250px;
	>.title {
		font-size: 1.2em;
	}
	>.description {
		white-space: normal;
		line-height: 1.2em;
		color: rgba(0, 0, 0, 0.5);
	}
}
.ellipsis {
	white-space: nowrap;
	overflow: hidden;
	text-overflow: ellipsis;
}
