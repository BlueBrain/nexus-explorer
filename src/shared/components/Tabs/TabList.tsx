import * as React from 'react';
import './TabList.less';
import { Tabs } from 'antd';
import TabElement from './TabElement';

const { TabPane } = Tabs;



type TabItem = {
  id: string;
  label: string;
  description: string;
};

type TabListProps = {
  items: TabItem[];
  onSelected: (activeKey: string) => void;
  defaultActiveId?: string;
  position?: 'left' | 'right' | 'top' | 'bottom' | undefined;
  onEditClick? : ( id : string) => void 
};



const TabList: React.FunctionComponent<TabListProps> = ({
  items,
  onSelected,
  defaultActiveId,
  position = 'left',
  children,
  onEditClick
}) => {
  const editClick = onEditClick?  onEditClick :  ( id: string ) => {};
  return (
    <div className="tab-list">
      <Tabs
        tabPosition={position}
        onChange={onSelected}
        defaultActiveKey={defaultActiveId}
      >
        {items.map(({ label, description, id }) => (
          <TabPane
            tab={
              <TabElement label={label} description={description} id={id} onEditClick={editClick}/>
            }
            key={id}
          >
            {children}
          </TabPane>
        ))}
      </Tabs>
    </div>
  );
};

export default TabList;
