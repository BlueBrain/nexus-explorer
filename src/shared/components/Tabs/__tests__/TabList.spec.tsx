import * as React from 'react';
import { shallow } from 'enzyme';

// Need to mock Date.now() because it is used down the line
// to generate a per-element class, which breaks snapshot testing.
// Must be done before importing the React component.
jest.spyOn(Date, 'now').mockImplementation(() => 1234567890123);

import TabList from '../TabList';

const tabListCompenent = <TabList items={[]} onSelected={jest.fn}/>;
const wrapper = shallow(tabListCompenent);

describe('TabList component', () => {
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

// Restoring Date.now() to the real thing, so as not to break other tests.
jest.restoreAllMocks();
