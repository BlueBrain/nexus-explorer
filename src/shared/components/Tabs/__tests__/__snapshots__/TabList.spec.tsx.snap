// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabList component should render correctly 1`] = `
<div
  className="tab-list"
>
  <Tabs
    onChange={[Function]}
    tabPosition="left"
  />
</div>
`;
